crates/biome_js_analyze/src/lint/nursery/no_static_element_interactions.rs

```diff
@@ -0,0 +1,353 @@
+                }
+            };
+        }
+
+        Some(())
+    }
+
+    fn diagnostic(ctx: &RuleContext<Self>, _: &Self::State) -> Option<RuleDiagnostic> {
+        let node = ctx.query();
+        Some(RuleDiagnostic::new(
+            rule_category!(),
+            node.range(),
+            markup! {{"Avoid non-native interactive elements. If using native HTML is not possible, add an appropriate role and support for tabbing, mouse, keyboard, and touch inputs to an interactive content element"}},
+        ))
+    }
+}
+
+/// Check if any interactive handler is present in the attributes.
+fn is_interactive_handler_present(attributes: &FxHashMap<String, Vec<String>>) -> bool {
+    let categories_to_check = vec!["focus", "keyboard", "mouse"];
+    for category in categories_to_check {
+        if let Some(handlers) = INTERACTIVE_HANDLER_LIST.get(category) {
+            for handler in handlers {
+                if let Some(values) = attributes.get(*handler) {
+                    if values.iter().any(|value| value != "null") {
+                        return true;
+                    }
+                }
+            }
+        }
+    }
+    false
+}
+
+/// This method is an override of AriaService::extract_attributes.
+///
+/// This modified version can extract attributes with dynamic values as well.
+/// The original extract_attributes in AriaService skips extracting other static attributes if a dynamic value is encountered.
+pub fn extract_attrs(attribute_list: &JsxAttributeList) -> Option<FxHashMap<String, Vec<String>>> {
+    let mut defined_attributes: FxHashMap<String, Vec<String>> = FxHashMap::default();
+    for attribute in attribute_list {
+        if let AnyJsxAttribute::JsxAttribute(attr) = attribute {
+            let name = attr.name().ok()?.syntax().text_trimmed().to_string();
+            let values = if let Some(initializer) = attr.initializer() {
+                let initializer = initializer.value().ok()?;
+                if let Some(static_value) = initializer.as_static_value() {
+                    static_value
+                        .text()
+                        .split(' ')
+                        .map(|s| s.to_string())
+                        .collect::<Vec<String>>()
+                } else {
+                    vec![initializer.syntax().text().to_string()]
+                }
+            } else {
+                vec!["true".to_string()]
+            };
+
+            defined_attributes.entry(name).or_insert(values);
+        }
+    }
+    Some(defined_attributes)
+}
+
+fn has_truthy_ariahidden_attr(node: &AnyJsxElement) -> bool {
+    return node
+        .find_attribute_by_name("aria-hidden")
+        .map_or(false, |attr| {
+            attr.as_static_value()
+                .map_or(true, |val| val.text() == "true")
+        });
+}
+
+/// This function allows non-interactive Roles.
+/// That is because these are Roles enabled in eslint-plugin-jsx-a11y.
+fn has_valid_role(role: &str) -> bool {
```

Comment:
casually asking; Is there any reason to add this function instead of using `aria_roles.is_role_interactive()`?

Reply:
@unvalley 
Thank you for your comment! 

You are right, I initially considered using that method. 
However, there was a discrepancy between the test cases in eslint-plugin-jsx-a11y and the definition of interactive roles in Biome. 

Upon investigation, I found that Biome's role definitions are correct and should be treated as invalid, but eslint-plugin-jsx-a11y treats them as valid cases. 
Therefore, I created a custom method.

Reply:
> Upon investigation, I found that Biome's role definitions are correct and should be treated as invalid, but eslint-plugin-jsx-a11y treats them as valid cases.

If so, I think the test case should be updated. W3C's document is more reliable.

Reply:
@unvalley 
fixed in this commit ae05ccaebd8b2e24968e6d1542ff7a94282663dc

Reply:
@unvalley 
Can I this conversation  be resolved?